### PR TITLE
Fix PHP 5.6 / 7.1 compatibility

### LINKED ISSUE #31: Latest version (1.0.2) can't be used on PHP 5.6 and 7.0
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | 1.0.2

#### Summary

@weierophinney As discussed yday on Slack the latest version (1.0.2) cannot be used on PHP 5.6 and 7.0 because of added ocramius/package-version dependency which requires PHP 7.1+

#### Expected behaviour

We can use migration tool on any PHP version starting from 5.6.

### PR DESCRIPTION
**Pull request by tux-rampage**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

This PR aims to fix the Regression #31 (BC break).

It removes ocramius/package-versions and provides a local alternative that is compatible with php 5.6+

### CODE REVIEW

File: src/PackageVersions.php [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+use function is_object;
+use function is_string;
+use function json_decode;
+
+/**
+ * @internal
+ */
+final class PackageVersions
+{
+    const COMPOSER_INSTALLED = __DIR__ . '/../../../../composer/installed.json';
```
Comment (tux-rampage) [inline src/PackageVersions.php:28]: Getting the own package version only works on the root package's files and when the package was installed as dependency there.

Therefore only these paths are checked.

File: bin/laminas-migration [hunk 2]
```diff
@@ -23,9 +22,7 @@ if (file_exists($a = __DIR__ . '/../../../autoload.php')) {
     exit(1);
 }
 
-$version = strstr(Versions::getVersion('laminas/laminas-migration'), '@', true);
-
-$application = new Application('laminas-migration', $version);
+$application = new Application('laminas-migration', PackageVersions::getAppVersion());
```
Comment (tux-rampage) [inline bin/laminas-migration:25]: This seems to be the only use case for `PackageVersions`, therefore i stripped it down to this one.

**Comment by tux-rampage** [comment]
There is still a bug, will add test cases and fix later.

**Comment by tux-rampage** [comment]
Test-Cases and fix added for the following edge-cases:

- both of the composer files are not present
- an empty list is provided to `PackageVersions::fromComposerFiles()`

Both cases did result in a failure of `array_merge()`

**Review by weierophinney — APPROVED** [review]
:ship:
